chore: Remove themes submenu from the Appearance menu

nickcernis opened the pull request:
For https://wpengine.atlassian.net/browse/BH-791.

## Acceptance Criteria
- Remove Themes and all related items from the WP navigation
- Prevent navigation directly to theme pages

## Notes
Changing themes is still possible via WP-CLI or by manually entering the Customizer URL ( `http://example.com/wp-admin/customize.php` ) and clicking “Change” next to “Active theme” in the Customizer root.

Not sure we need to block the Customizer theme page since we also remove Customizer links by default, but noting in case.

nickcernis committed:
chore: remove Appearance → Themes submenu page

mindctrl commented:
Looks good. Thoughts on removing Themes from the dropdown in the admin menu?

Screenshot
![image](https://user-images.githubusercontent.com/1254870/102899548-15882280-4439-11eb-8154-31019947e56d.png)

nickcernis committed:
chore: remove Themes admin bar link

nickcernis commented:
> Looks good. Thoughts on removing Themes from the dropdown in the admin bar?

@mindctrl Good idea, thanks! Done in https://github.com/wpengine/headless-framework/pull/50/commits/a9de9e06fb39eaa2fee8e3c47ab4b09e6c748d8e.

For others, this requires commenting out this line in `wpe-headless.php` to view the admin bar on the front end to test it:

```php
require WPE_HEADLESS_DIR . '/includes/deny-public-access/callbacks.php';
```

nickcernis committed:
chore: prevent access to theme admin page URL

Attempts to visit http://example.com/wp-admin/themes.php
will redirect to http://example.com/wp-admin/.

nickcernis commented:
As discussed in standup, attempts to visit `wp-admin/themes.php` by URL will not redirect to `wp-admin/`:

https://github.com/wpengine/headless-framework/pull/50/commits/6502f1afcc73da7d48b79eca254321bf8d8d39d7
